avm/res/azure-stack-hci/logical-network/tests/e2e/waf-aligned/main.test.bicep

```diff
@@ -0,0 +1,219 @@
+        }
+        {
+          adapter: ['comp0', 'comp1']
+          name: 'compute'
+          overrideAdapterProperty: true
+          adapterPropertyOverrides: {
+            jumboPacket: '9014'
+            networkDirect: 'Disabled'
+            networkDirectTechnology: 'iWARP'
+          }
+          overrideQosPolicy: false
+          qosPolicyOverrides: {
+            bandwidthPercentage_SMB: '50'
+            priorityValue8021Action_Cluster: '7'
+            priorityValue8021Action_SMB: '3'
+          }
+          overrideVirtualSwitchConfiguration: false
+          virtualSwitchConfigurationOverrides: {
+            enableIov: 'true'
+            loadBalancingAlgorithm: 'Dynamic'
+          }
+          trafficType: ['Compute']
+        }
+        {
+          adapter: ['smb0', 'smb1']
+          name: 'storage'
+          overrideAdapterProperty: true
+          adapterPropertyOverrides: {
+            jumboPacket: '9014'
+            networkDirect: 'Disabled'
+            networkDirectTechnology: 'iWARP'
+          }
+          overrideQosPolicy: true
+          qosPolicyOverrides: {
+            bandwidthPercentage_SMB: '50'
+            priorityValue8021Action_Cluster: '7'
+            priorityValue8021Action_SMB: '3'
+          }
+          overrideVirtualSwitchConfiguration: false
+          virtualSwitchConfigurationOverrides: {
+            enableIov: 'true'
+            loadBalancingAlgorithm: 'Dynamic'
+          }
+          trafficType: ['Storage']
+        }
+      ]
+      storageConnectivitySwitchless: false
+      storageNetworks: [
+        {
+          adapterName: 'smb0'
+          vlan: '711'
+        }
+        {
+          adapterName: 'smb1'
+          vlan: '712'
+        }
+      ]
+      subnetMask: '255.255.255.0'
+      driftControlEnforced: true
+      smbSigningEnforced: true
+      smbClusterEncryption: true
+      sideChannelMitigationEnforced: true
+      bitlockerBootVolume: true
+      bitlockerDataVolumes: true
+    }
+    tags: {
+      'hidden-title': 'This is visible in the resource name'
+      Environment: 'Non-Prod'
+      Role: 'DeploymentValidation'
+    }
+  }
+}
+
+resource customLocation 'Microsoft.ExtendedLocation/customLocations@2021-08-31-preview' existing = {
+  scope: resourceGroup
+  name: '${namePrefix}${serviceShort}-location'
+  dependsOn: [
+    azlocal
+  ]
+}
+
+module testDeployment '../../../main.bicep' = {
+  name: '${uniqueString(deployment().name, enforcedLocation)}-logicalNetwork-${serviceShort}'
+  scope: resourceGroup
+  params: {
+    name: '${namePrefix}${serviceShort}logicalnetwork'
+    location: enforcedLocation
+    customLocationId: customLocation.id
+    vmSwitchName: 'ConvergedSwitch(management)'
+    ipAllocationMethod: 'Static'
+    addressPrefix: '172.20.0.1/24'
+    startingAddress: '172.20.0.171'
+    endingAddress: '172.20.0.190'
+    defaultGateway: '172.20.0.1'
+    dnsServers: ['172.20.0.1']
+    routeName: 'default'
+    vlanId: null
```

Comment:
Similar comment to the one in the defaults test. Are all these parameters needed to show how to deploy the module in a WAF-aligned way?
I think what you want is a 3rd test cases that sets these values (usually something we refer to as the `max` test)

Reply:
Note, that `max` test should then also contain test for e.g. the role assignments (which is something you can copy from other modules in this repository)